src/core/numericformats/qgsbasicnumericformat.h

```diff
@@ -0,0 +1,133 @@
+/***************************************************************************
+                             qgsbasicnumericformat.h
+                             --------------------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSBASICNUMERICFORMAT_H
+#define QGSBASICNUMERICFORMAT_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsnumericformat.h"
+#include <sstream>
+#include <iostream>
+#include <memory>
+
+/**
+ * \ingroup core
+ * A numeric formatter which returns a simple text representation of a value.
+ *
+ * \since QGIS 3.12
+ */
+class CORE_EXPORT QgsBasicNumericFormat : public QgsNumericFormat
+{
+  public:
+
+    /**
+      * Default constructor
+      */
+    QgsBasicNumericFormat();
+
+    //! QgsBasicNumericFormat cannot be copied
+    QgsBasicNumericFormat( const QgsBasicNumericFormat & ) = delete;
+    //! QgsBasicNumericFormat cannot be copied
+    QgsBasicNumericFormat &operator=( const QgsBasicNumericFormat & ) = delete;
+
+    QString id() const override;
+    QString formatDouble( double value, const QgsNumericFormatContext &context ) const override;
+    QgsNumericFormat *clone() const override SIP_FACTORY;
+    QgsNumericFormat *create( const QVariantMap &configuration ) const override SIP_FACTORY;
```

Comment:
isn't it static?

Reply:
I'm using the prototype pattern here, so it needs to be a virtual member